protocol_rfcs/redirection.md

```diff
@@ -0,0 +1,116 @@
+# Redirection
+**Associated Github issue for discussions: https://github.com/delta-io/delta/issues/3702**
+<!-- Remove this: Replace XXXX with the actual github issue number -->
+<!--  Give a general description / context of the protocol change, and remove this comment. -->
+This RFC proposes a new kind of features called "redirection" for Delta tables. The feature enables seamless migration of existing Delta tables to different storage locations while minimizing disruption to user workflows. This enhancement aims to reduce downtime and simplify the process of relocating Delta tables.
+## Terminology:
+* Modern Delta clients: These are Delta clients that are compatible with the latest Delta protocol and support the redirection feature.
+* Legacy Delta clients: These refer to Delta clients that do not support the redirection feature, as they are not up-to-date with the latest Delta protocol.
+* Redirect Source Table: The source Delta table that is redirected to a new storage location.
+* Redirect Destination Table: The Delta table that is created as a redirected table of an existing source table. Any access to the redirect source table would be dispatched to this redirect destination table.
+## Overview
+The proposal includes two distinct table features:
+* **RedirectReaderWriter**: This feature enables both read and write operations to be redirected from the source to the destination for Modern Delta clients. It blocks all read and write operations from Legacy Delta clients.
```

Comment:
can we we remove this term "modern" from everywhere? really ambiguous term... what modern compared to what?